samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.system.inmemory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.apache.samza.Partition;
+import org.apache.samza.system.EndOfStreamMessage;
+import org.apache.samza.system.IncomingMessageEnvelope;
+import org.apache.samza.system.StreamSpec;
+import org.apache.samza.system.SystemStream;
+import org.apache.samza.system.SystemStreamMetadata;
+import org.apache.samza.system.SystemStreamPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Initial draft of in-memory manager. It is test only and not meant for production use right now.
+ */
+public class InMemoryManager {
+  private static final Logger LOG = LoggerFactory.getLogger(InMemoryManager.class);
+  private static final int DEFAULT_PARTITION_COUNT = 1;
+
+  private final ConcurrentHashMap<SystemStreamPartition, List<IncomingMessageEnvelope>> bufferedMessages;
+  private final Map<SystemStream, Integer> systemStreamToPartitions;
+
+  public InMemoryManager() {
+    bufferedMessages = new ConcurrentHashMap<>();
+    systemStreamToPartitions = new ConcurrentHashMap<>();
+  }
+
+  private List<IncomingMessageEnvelope> newSynchronizedLinkedList() {
+    return  Collections.synchronizedList(new LinkedList<IncomingMessageEnvelope>());
+  }
+
+  public void put(SystemStreamPartition ssp, Object message) {
```

Comment:
no need for this function. Use the following one with a null check or something.